src/Microsoft.OData.Core/DuplicatePropertyNameChecker.cs

```diff
@@ -15,12 +15,13 @@ namespace Microsoft.OData
     ///   2) No duplicate "@odata.associationLink" on a property.
     ///   3) "@odata.associationLink"s are put on navigation properties.
     /// </summary>
-    internal class DuplicatePropertyNameChecker : IDuplicatePropertyNameChecker
+    internal class DuplicatePropertyNameChecker : IDuplicatePropertyNameChecker, IDisposable

```

Comment:
I haven't seen the `Dispose()` method or a `using` statement being used anywhere for objects of this type. How is the `IDisposable` applied for this type?

When getting the item from the pool, you're calling `duplicateNameChecker.Reset()` manually, which is the only thing that dispose seems to be doing. Not sure where else dispose is being applied.

Reply:
#Resolved